buffer/src/main/java/io/netty/buffer/SlicedByteBuf.java

```diff
@@ -92,8 +86,13 @@ public boolean isDirect() {
     }
 
     @Override
-    public int capacity() {
-        return length;
+    public final int capacity() {
```

Comment:
@normanmaurer - I pushed a commit which achieves the same goal without breaking the existing API ... ptal